### DISCUSSION TITLE
Select * behaves differently in 1.4 (appears to work in 1.3/2.0)

### DISCUSSION THREAD

**Discussion by DanCardin** [sqlalchemy/sqlalchemy#11323] [Usage Questions] [answered]
Hi, I'm trying to support sqlalchemy 1.3+ in a plugin, and am struggling to produce a query that acts the same in all versions (or more specifically, that works correctly at all in 1.4)

```python
table = metadata.tables[tablename]

query = select(text("*")).select_from(table).where(or_(*filters))
result = connection.execute(query).fetchall()
```

At least with the code as it exists today, I need literal `*` here because (in the middle of an alembic migration) the `table` object here (and the sqlalchemy `filters`) might not reflect the actual set of table columns that exist in the current moment.

In both 1.3, and 2.0, this does what it's supposed to, from which I can
```
for record in result:
    record_dict = row_to_dict(record)
```

In 1.4 only, it produces:
```
lib/python3.8/site-packages/sqlalchemy/engine/result.py:1072: in fetchall
    return self._allrows()
lib/python3.8/site-packages/sqlalchemy/engine/result.py:403: in _allrows
    made_rows = [make_row(row) for row in rows]
>   made_rows = [make_row(row) for row in rows]
E   RuntimeError: number of values in row (2) differ from number of column processors (1)
```

If there's an obvious way to make it work **just** in 1.4, i'll be happy to split behavior on that. otherwise, ideally the same code would work across the board.

---

Up to now, I've been circumventing this issue by compiling the query string and executing that inside a `text()`, but it's starting to run afoul of double-escaping issues that seem like they're just a mess compared to this being so close to working (obviously except 1.4)

**Comment by CaselIT** [comment]
Hi,

Does using `select('*')...` (without the `text()`) work on 1.4 in your use case?

**Reply by DanCardin** [reply]
No :( , same error

**Comment by gordthompson** [comment, marked as answer]
This works for me:

```python
# fmt: off
from sqlalchemy import Column, Integer, MetaData, String, Table, create_engine, select, text, __version__ as ver
# fmt: on

print(ver)  # 1.4.52

engine = create_engine("sqlite://")

table = Table(
    "t",
    MetaData(),
    Column("id", Integer, primary_key=True),
    Column("txt", String),
)
table.create(engine)

with engine.begin() as conn:
    conn.exec_driver_sql("INSERT INTO t (txt) VALUES ('foo')")

query = select(text("*")).select_from(table).where(table.c.txt == "foo")


def row_to_dict(r):
    return r._mapping


with engine.begin() as connection:
    result = connection.execute(query).fetchall()
    for record in result:
        record_dict = row_to_dict(record)
        print(record_dict)
    """
    {'id': 1, 'txt': 'foo'}
    """
```

**Reply by DanCardin** [reply]
hmm same. I'm hard struggling to reproduce it in this simplified sort of example, whether with sqlite or postgres.

It seems that it must be something to do with the local `table` definition having a different number of columns than the actual table, as that's the actual failing test case here: https://github.com/DanCardin/sqlalchemy-declarative-extensions/actions/runs/8839704397/job/24273620924?pr=56. More specifically the **actual** table has an additional column that the metadata definition does not have.

But I've gone pretty far down the rabbit hole of trying to make the example situation here increasingly like the code in the test, but it continues to work 😮‍💨

**Reply by DanCardin** [reply]
I have no idea what's going on. It's something to do with how sqlalchemy events are being registered in my test case and...idk something else. Without the event registration, it works fine. with it, it fails as described.

ultimately I really feel like it's a test-case-specific scenario that i've managed to adjust enough to work, and that it's (hopefully) not an issue with the actual compatibility in the the actual code.

Your passing testcase set me down this investigative path, so you get the answer bubble :P ! Thanks!

**Comment by zzzeek** [comment]
hi -

I can't reproduce this.  can you send a reproduction case and what you are doing for `row_to_dict()`?  the below runs fine on 1.4  thanks 

```py

from sqlalchemy import Column
from sqlalchemy import create_engine
from sqlalchemy import Integer
from sqlalchemy import MetaData
from sqlalchemy import or_
from sqlalchemy import select
from sqlalchemy import Table
from sqlalchemy import table
from sqlalchemy import text

table = Table(
    "mytable", MetaData(),
    Column("x", Integer),
    Column("y", Integer),

)

e = create_engine("sqlite://", echo='debug')

table.create(e)

with e.connect() as conn:
    conn.execute(
        table.insert().values(
            [{"x":1, "y": 2}, {"x":2, "y": 3}, {"x":3, "y": 4}]
        )
    )


    query = select(text("*")).select_from(table).where(or_(table.c.x >1, table.c.y < 3))
    result = conn.execute(query).fetchall()


    for record in result:
        record_dict = dict(record)  # note this is not 2.0 compatible in any case
```
